src/main/java/teammates/ui/webapi/WebApiServlet.java

```diff
@@ -149,4 +153,28 @@ private void throwError(HttpServletResponse resp, int statusCode, String message
         result.send(resp);
     }
 
+    private void putUserInfo(Map<String, Object> responseDetails, Action action) {
+        if (action == null) {
+            return;
+        }
+
+        String googleId = action.userInfo == null ? null : action.userInfo.getId();
+        String regkey = action.getRequestParamValue(Const.ParamsNames.REGKEY);
+        String studentEmail = action.getRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
```

Comment:
Not even `getRequestParamValue` should be accessible. This is already part of the internal business logic of the action class.
You'll soon find that this will not be allowed anymore.

Reply:
Understood. It is updated.